opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -2037,7 +2041,8 @@ namespace Opm
         switch(currentGroupControl) {
         case Group::InjectionCMode::NONE:
         {
-            OPM_DEFLOG_THROW(std::runtime_error, "NONE group control not implemented for injectors" , deferred_logger);
+            // The NONE case is handled earlier
+            assert(false);
```

Comment:
What is this supposed to do?  `Assert(false)` vanishes when `NDEBUG` is defined (i.e., in release mode).

Reply:
This code part should never be called since the NONE case is handled just above as a special case. The assert is just to give a snap to the developer if it tries to remove the NONE handling above.

Reply:
> The assert is just to give a snap to the developer if it tries to remove the NONE handling above.

That's my point.  Developers who typically use release builds wont see `assert(false)` at all.

Reply:
The assert() is an acceptable way to guard against developer error, even if release mode builds do not see it. I'll let this stand.